src/solvers/smt2/smt2_conv.cpp

```diff
@@ -4869,8 +4869,10 @@ void smt2_convt::find_symbols_rec(
 
     if(recstack.find(id) == recstack.end())
     {
+      const auto &base_struct = ns.follow_tag(struct_tag);
       recstack.insert(id);
-      find_symbols_rec(ns.follow_tag(struct_tag), recstack);
+      find_symbols_rec(base_struct, recstack);
+      datatype_map[type] = datatype_map[base_struct];
     }
```

Comment:
:+1: I have tested this locally and this does indeed seem to fix a failure wherein cbmc prints `map::at` and exits.

Reply:
Thanks.